### PR TITLE
add better error message for too-long URI

### PR DESCRIPTION
**Pull request by mfeurer**
In response to https://github.com/openml/openml-python/issues/468

### CODE REVIEW

File: openml/_api_calls.py [hunk 1]
```diff
@@ -159,9 +231,12 @@ def _parse_server_exception(
     try:
         server_exception = xmltodict.parse(response.text)
     except Exception:
-        raise OpenMLServerError(
-            'Unexpected server error when calling {}. Please contact the developers!\n'
-            'Status code: {}\n{}'.format(url, response.status_code, response.text))
+        if response.status_code == 414:
```
Comment (PGijsbers) [inline openml/_api_calls.py]: Does it not make sense to add this before the try/except block? I don't know the details, but sounds like you would always assume the parsing to fail if the response had code `414`.

**Comment by PGijsbers** [comment]
It looks like there is no unit test verifying these long uris now raise a descriptive error?

File: openml/_api_calls.py (hunk 1 above)
Comment (mfeurer) [reply openml/_api_calls.py]: > Does it not make sense to add this before the try/except block?

Yes, it does.

> but sounds like you would always assume the parsing to fail if the response had code 414.

No, a lot of things can happen with the server output...

**Comment by mfeurer** [comment]
I just added a test.

**Comment by PGijsbers** [comment]
Approved and merged. Should we add a guideline not to move functions around? It's hard to spot what changes, if any, are made in a moved function as it is caught in the diff regardless). Unless it's a specific refactor PR or the dev explicitly says which lines have been changed (if any) for each moved function? Generally speaking, moving functions around won't (shouldn't) happen often, and the overhead introduced for the dev to point out changes (if any) is far smaller than the overhead for the reviewer trying to identify where changes are made (if any).

**Comment by mfeurer** [comment]
That makes sense. Would you like to add this to the contribution guidelines (maybe finishing up the PR #689 on the way)?

**Pull request merged**